### PR TITLE
Improve form test coverage

### PR DESCRIPTION
**Pull request by backus**
These changes bring the `Form` coercions to 100% mutation coverage

### CODE REVIEW

**Comment by backus** [comment]
Ah looks like I need to rebase these changes

File: lib/dry/types/coercions/form.rb [hunk 1]
```diff
@@ -10,13 +10,10 @@ module Form
         TRUE_VALUES = %w[1 on  t true  y yes].freeze
         FALSE_VALUES = %w[0 off f false n no].freeze
         BOOLEAN_MAP = ::Hash[TRUE_VALUES.product([true]) + FALSE_VALUES.product([false])].freeze
+        EMPTY_STRING = ''.freeze
```
Comment (solnic) [inline lib/dry/types/coercions/form.rb]: Hah literally yesterday I looked at these coercions and noticed this duplicated empty string allocations and thought "ugh, this should've been a constant" :)

**Comment by solnic** [comment]
Could rebase this? I merged in new JSON types 30 seconds before you sent this PR and we have conflicts.

**Comment by backus** [comment]
> Could rebase this? I merged in new JSON types 30 seconds before you sent this PR and we have conflicts.

already on it :)

**Comment by backus** [comment]
@solnic rebased

**Comment by backus** [comment]
The earlier builds here fail because I use [`git push-each`](https://github.com/dkubb/git-tools/blob/master/git-push-each) and travis does not checkout specific commit SHAs when github triggers a build on their platform. As a result the earlier commits here are actually the results of builds that were fixed when I rebased and force pushed recently. 

tl;dr sorry about the earlier failing commits I think you can safely ignore them
